handlers.go

```diff
@@ -265,10 +265,10 @@ func (dht *IpfsDHT) handleFindPeer(ctx context.Context, from peer.ID, pmes *pb.M
 	if targetPid == dht.self {
 		closest = []peer.ID{dht.self}
 	} else {
-		closest = dht.betterPeersToQuery(pmes, from, dht.bucketSize)
-
 		// Never tell a peer about itself.
 		if targetPid != from {
+			closest = dht.betterPeersToQuery(pmes, from, dht.bucketSize)
+
```

Comment:
This might be ok and save on some bandwidth in the even a peer does look up itself, but we already don't send a peer back to itself even when calling `dht.betterPeersToQuery`.

https://github.com/libp2p/go-libp2p-kad-dht/blob/03d4b62da366249c37dbb829c869212fbafba734/dht.go#L694

This prevents us from sending back the closest peers which is part of the general query contract. It doesn't mean we can't do this optimization, but it just might not be worthwhile.

Reply:
ok, you can ignore it.
The number of peers in my test network is less than K (closest peeers), so [dht.routingTable.NearestPeers](https://github.com/libp2p/go-libp2p-kad-dht/blob/03d4b62da366249c37dbb829c869212fbafba734/query.go#L151) may contains itself.